### PR TITLE
New endpoints Major Release Update

### PR DESCRIPTION
**Pull request by RobDWaller**
- Adds a number of new endpoints to provide more functionality.
- Now enforces secret security policies
- No longer supports php5.6
- Added type hinting
- Extended tests
- Improved comments
- Added additional info to Readme
- Added license file, MIT

### CODE REVIEW

File: src/Helper/Base64UrlConverter.php [hunk 1]
```diff
@@ -1,33 +1,76 @@
...
 
-    public function setBase64UrlString($base64UrlString)
+    /**
+     * Set the base64 Url string
+     *
+     * @param string $base64UrlString
+     *
+     * @return self
+     */
+    public function setBase64UrlString($base64UrlString): self
```
Comment (gemmadlou) [inline src/Helper/Base64UrlConverter.php]: Possibly typehint the parameter as a string?

File: src/TokenBuilder.php [hunk 2]
```diff
@@ -215,16 +245,27 @@ public function setIssuer($issuer)
...
+        }
+
+        throw new TokenBuilderException('Failed to add payload, format wrong. Array must contain key and value.');
     }
 
-    private function encodeHeader()
+    /**
+     * Encode the header string and return it
+     *
+     * @return string
```
Comment (gemmadlou) [inline src/TokenBuilder.php]: Would you use @throws phpdoc tag in your docblock comments where it's possible to get exceptions?

File: src/Helper/Base64UrlConverter.php (hunk 1 above)
Comment (RobDWaller) [reply src/Helper/Base64UrlConverter.php]: Good spot, I will change.

File: src/TokenBuilder.php (hunk 2 above)
Comment (RobDWaller) [reply src/TokenBuilder.php]: Another good spot.

File: src/Helper/Base64UrlConverter.php [hunk 3]
```diff
@@ -45,26 +93,46 @@ private function convertToBase64()
...
 
         return $this;
     }
 
+    /**
+     * Trigger base64 Url string to base64 string process
+     *
+     * @return self
+     */
     public function toBase64()
```
Comment (RobDWaller) [inline src/Helper/Base64UrlConverter.php]: fix return type

**Pull request merged**